### PR TITLE
[Unity] Add new extractor (fixes issue #14528)

### PR DESCRIPTION
**Pull request by parmjitv**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

Added a new extractor for the Unity platform [https://unity3d.com/] - issue #14528.

This site provides a number of free video tutorials for Unity, a popular content creator for 3D games and apps.

Cheers and thank you!

Parmjit V.

### CODE REVIEW

File: youtube_dl/extractor/unity.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        youtube_id = self._search_regex(
+            r'data-video-id="([_0-9a-zA-Z-]+)"',
+            webpage, 'youtube ID', default=None)
+        if not youtube_id:
+            raise ExtractorError('Unable to extract youtube ID', expected=True)
```
Comment (dstftw) [inline youtube_dl/extractor/unity.py]: This is the behavior of `_search_regex` with defaults.

File: youtube_dl/extractor/unity.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        youtube_url = 'https://youtu.be/%s' % youtube_id
+        return self.url_result(youtube_url, ie=YoutubeIE.ie_key())
```
Comment (dstftw) [inline youtube_dl/extractor/unity.py]: 1. Youtube id can be passed directly.
2. `video_id=` should be passed as well.

**Comment by parmjitv** [comment]
@dstftw - Done, and done.  Please let me know if there are any further corrections that need to be made.  Cheers!

File: youtube_dl/extractor/unity.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        vid = self._match_id(url)
+        webpage = self._download_webpage(url, vid)
+        youtube_id = self._search_regex(
+            r'data-video-id="([_0-9a-zA-Z-]+)"',
+            webpage, 'youtube ID')
+        return self.url_result(youtube_id, video_id=vid, ie=YoutubeIE.ie_key())
```
Comment (dstftw) [inline youtube_dl/extractor/unity.py]: Despite being keyword arguments avoid changing the original order.

File: youtube_dl/extractor/unity.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            'uploader_id': 'Unity3D',
+            'upload_date': '20140926',
+        }
+    }, {
+        'url': 'https://unity3d.com/learn/tutorials/projects/2d-ufo-tutorial/following-player-camera?playlist=25844',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        vid = self._match_id(url)
```
Comment (dstftw) [inline youtube_dl/extractor/unity.py]: `video_id` is idiomatic.

**Comment by parmjitv** [comment]
@dstftw - Gotcha.  I have reverted the var name and set the proper order for the args.  Please let me know if there is anything else left outstanding.  Thanks!

File: youtube_dl/extractor/unity.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        youtube_id = self._search_regex(
+            r'data-video-id="([_0-9a-zA-Z-]+)"',
+            webpage, 'youtube ID')
+        return self.url_result(youtube_id, YoutubeIE.ie_key(), video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/unity.py]: Keep keywords.

**Comment by parmjitv** [comment]
@dstftw - Done!  Anything else to be corrected?

**Pull request merged**